ext/CatalystBifurcationKitExtension/bifurcation_kit_extension.jl

```diff
@@ -13,7 +13,7 @@ function BK.BifurcationProblem(rs::ReactionSystem, u0_bif, ps, bif_par, args...;
 
     # Creates NonlinearSystem.
     Catalyst.conservationlaw_errorcheck(rs, vcat(ps, u0))
-    nsys = convert(NonlinearSystem, rs; remove_conserved=true, defaults=Dict(u0))
+    nsys = complete(convert(NonlinearSystem, rs; remove_conserved=true, defaults=Dict(u0)))
```

Comment:
Both `XProblem` and `convert(XSystem, rs)` require the output to be complete. In the extensions, if the input `rs` is not complete, the `convert` call should throw an error. Next, because `convert(NonlinearSystem, rs; ...)` always yield an incomplete system, this has to be completed to be passed into `NonlinearProblem` or later on.

I think this bit is analogue e.g. `XProblem` first passing `rs` to `xsys = convert(XSystem, rs)` (which throw an error if `rs` is not complete), and then internally marks the output `x = complete(xsys).

I will add some tests to the extensions to ensure they throw an error if an input system is not complete, I agree this should not be allowed.